rn/Teacher/ios/Submissions/PostSettingsViewController.swift

```diff
@@ -0,0 +1,142 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2019-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import UIKit
+import Core
+
+class PostSettingsViewController: UIViewController {
+    @IBOutlet weak var menu: HorizontalMenuView!
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var menuHeight: NSLayoutConstraint!
+    private var postGradesViewController: PostGradesViewController!
+    private var hideGradesViewController: HideGradesViewController!
+
+    enum MenuItem: Int {
+        case post, hide
+    }
+
+    static func create() -> PostSettingsViewController {
+        let controller = loadFromStoryboard()
+        return controller
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = NSLocalizedString("Post Settings", comment: "")
+        configureMenu()
+        configurePost()
+        configureHide()
+        scrollView.delegate = self
+    }
+
+    func configureMenu() {
+        menu.delegate = self
+        menuHeight.constant = 1.0 / UIScreen.main.scale
+    }
+
+    func configurePost() {
+        postGradesViewController = PostGradesViewController.create()
+        embed(postGradesViewController, in: scrollView) { child, _ in
+            child.view.addConstraintsWithVFL("H:|[view(==superview)]")
+            child.view.addConstraintsWithVFL("V:|[view(==superview)]|")
```

Comment:
what is `view(==superview)`? I find VFL really hard to read. I don't suppose any of this can be done in the storyboard?

Reply:
In VFL, view with ( `<some variable here>` ) could be a view of height or width 50  `i.e. view(50)` or an equality constraint saying "make this the same height or width as the superview"  `view(==superview)`.   

I think there is a time and place for storyboards and VFL. I find it pretty easy to read compared to storyboard xml ;).  Also, it's pretty nice knowledge to have when debugging auto layout when invalid constraints are present and one has to be broken to get your layout to work. When it comes to scroll views in storyboards, i much prefer VFL over storyboards.  ScrollViews in storyboards are much more difficult than they need to be IMO.  I do not like wasting time on layout if Interface builder if giving me the fits.

I did change this though to use storyboards.

Reply:
One day writing in SwiftUI will be much nicer than either IB or VFL.

Reply:
😂